picard/util/pipe.py

```diff
@@ -0,0 +1,206 @@
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+
+
+import concurrent.futures
+import os
+from sys import platform as PLATFORM
+
+
+if PLATFORM == "win32" or PLATFORM == "cygwin":
+    import win32pipe  # type: ignore
+    import win32file  # type: ignore
+    from pywintypes import error as WinApiError  # type: ignore
+
+    PLATFORM = "windows"
+
+
+class Pipe:
+    NO_RESPONSE_MESSAGE: str = "No response from FIFO"
+    NOT_FOUND_MESSAGE: str = "FIFO doesn't exist"
+    MESSAGE_TO_IGNORE: str = "Ignore this message, just testing the pipe"
+
+    def __init__(self, app_name: str, app_version: str, args=[]):
+        self.__app_name: str = app_name
+        self.__app_version: str = app_version
+        self.__platform: str = PLATFORM
+
+        self.path: str = self.__generate_filename()
+
+        self.is_pipe_owner: bool = False
+
+        # test if pipe is listened to even if no args provided
+        if type(args) == list:
+            if len(args) == 0:
+                args.append(self.MESSAGE_TO_IGNORE)
+        else:
+            raise ValueError("args argument MUST be a list")
+
+        if self.__platform == "windows":
+            for arg in args:
+                if not self.send_to_pipe(arg):
+                    self.is_pipe_owner = True
+                    break
+        else:
+            if self.__pipe_exists():
+                for arg in args:
+                    if not self.send_to_pipe(arg):
+                        os.unlink(self.path)
+                        self.__create_unix_pipe()
+                        break
+            else:
+                self.__create_unix_pipe()
+
+    def __pipe_exists(self) -> bool:
+        return os.path.exists(self.path)
+
+    def __generate_filename(self) -> str:
+        prefix: str = ""
+        username: str = os.getlogin()
+        if self.__platform == "windows":
+            prefix = "\\\\.\\pipe\\"
+        else:
+            prefix = "/tmp/"
+
+        return f"{prefix}{self.__app_name}_v{self.__app_version}_{username}_pipe_file"
+
+    def __create_unix_pipe(self) -> None:
+        os.mkfifo(self.path)
+        self.is_pipe_owner = True
+
+    def __win_sender(self, message: str) -> bool:
+        pipe = win32pipe.CreateNamedPipe(
+            self.path,
+            win32pipe.PIPE_ACCESS_DUPLEX,
+            win32pipe.PIPE_TYPE_MESSAGE | win32pipe.PIPE_WAIT,
+            1, 65536, 65536,
+            300,
```

Comment:
Those numerical values should be either constants (with explicit names) or a comment should be added to tell what they are.

Reply:
Created constants with links to docs just in case, hope it can do now